source/features/open-all-notifications.js

```diff
@@ -1,60 +1,102 @@
 import select from 'select-dom';
 import delegate from 'delegate';
-import {groupButtons} from '../libs/utils';
+import * as icons from '../libs/icons';
+import {groupButtons, safeElementReady} from '../libs/utils';
 import {isNotifications} from '../libs/page-detect';
 
-function openNotifications() {
-	const urls = select.all('.unread .js-notification-target').map(el => el.href);
+const unreadNotificationsClass = '.unread .js-notification-target';
+let notificationsToOpen = [];
+
+const faceboxTrigger = (
+	<a href="#open-all-in-tabs" id="facebox-trigger" rel="facebox" style={{display: 'none'}}></a>
+);
+
+function openNotifications(unreadNotifications = notificationsToOpen) {
+	const urls = unreadNotifications.map(el => el.href);
+
 	browser.runtime.sendMessage({
 		urls,
 		action: 'openAllInTabs'
 	});
-	for (const listItem of select.all('.list-group .list-group-item')) {
+
+	for (const notification of unreadNotifications) {
+		const listItem = notification.closest('.list-group-item');
 		listItem.classList.add('read');
+		listItem.classList.remove('unread');
 	}
 }
 
-export default function () {
-	if (!isNotifications() || select.exists('[href="#open_all_in_tabs"]')) {
-		return;
+function tryOpeningNotifications(container) {
+	notificationsToOpen = select.all(unreadNotificationsClass, container);
+
+	if (notificationsToOpen.length < 10) {
+		openNotifications();
+	} else {
+		faceboxTrigger.click();
 	}
-	const unreadCount = select.all('.unread .js-notification-target').length;
+}
 
-	if (unreadCount === 0) {
+export default function () {
+	if (!isNotifications() || select.exists('[href="#open-all-in-tabs"]')) {
 		return;
 	}
 
-	const openButton = <a href="#open_all_in_tabs" class="btn btn-sm">Open all unread in tabs</a>;
+	faceboxTrigger.addEventListener('click', async () => {
+		await safeElementReady('#facebox-description');
+
+		select('#facebox-description').textContent = `Are you sure you want to open ${notificationsToOpen.length} tabs?`;
+	});
+
+	document.body.append(
+		<div id="open-all-in-tabs" style={{display: 'none'}}>
+			{faceboxTrigger}
+
+			<h2 class="facebox-header" data-facebox-id="facebox-header">Are you sure?</h2>
+
+			<p data-facebox-id="facebox-description"></p>
+
+			<div class="full-button">
+				<button class="btn btn-block" id="open-all-notifications">Open all notifications</button>
+			</div>
+		</div>
+	);
+
+	delegate('#open-all-notifications', 'click', () => {
+		openNotifications();
+		select('.js-facebox-close').click(); // Close modal
+	});
 
-	// Make a button group
+	// Create an open button and add it into a button group
 	const group = select('.tabnav .float-right');
-	group.prepend(openButton);
+	group.prepend(
+		<a href="#open-all-in-tabs" id="open-all-tabs-trigger" class="btn btn-sm">Open all unread in tabs</a>
+	);
 	groupButtons([...group.children]);
 
-	// Move out the extra node that messes with .BtnGroup-item:last-child
-	document.body.append(select('#mark_as_read_confirm_box') || '');
```

Comment:
This line is needed

With it:

<img width="312" alt="before" src="https://user-images.githubusercontent.com/1402241/35932805-4dab82be-0c6b-11e8-9173-f641b2db5774.png">

Without it:

<img width="292" alt="after" src="https://user-images.githubusercontent.com/1402241/35932804-4d7ba0d0-0c6b-11e8-8ad0-0b2cec70d890.png">